tests/plugins/find_and_replace.test.ts

```diff
@@ -150,6 +156,84 @@ describe("basic search", () => {
     expect(matches).toHaveLength(2);
     expect(matchIndex).toStrictEqual(0);
   });
+
+  test("Need to update search if column or row is removed", () => {
+    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
+    let matches = model.getters.getSearchMatches();
+    let matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(4);
+    expect(matchIndex).toStrictEqual(0);
+    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ col: 0, row: 3, selected: false });
+    expect(matches[3]).toStrictEqual({ col: 0, row: 4, selected: false });
+    model.dispatch("REMOVE_COLUMNS_ROWS", {
+      dimension: "ROW",
+      elements: [1],
+      sheetId: model.getters.getActiveSheetId(),
+    });
+    matches = model.getters.getSearchMatches();
+    matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(3);
+    expect(matchIndex).toStrictEqual(0);
+    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ col: 0, row: 3, selected: false });
+  });
+
+  test("Need to update search if doing undo redo operation which can update the cell", () => {
+    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
+    let matches = model.getters.getSearchMatches();
+    let matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(4);
+    expect(matchIndex).toStrictEqual(0);
+    model.dispatch("REMOVE_COLUMNS_ROWS", {
```

Comment:
`deleteRow`

Reply:
Changes done and update the commit message